dynamiqs/time_tensor.py

```diff
@@ -239,3 +278,103 @@ def __add__(self, other: Tensor | TimeTensor) -> TimeTensor:
             return CallableTimeTensor(f, f0)
         else:
             return NotImplemented
+
+
+class _PWCTensor:
```

Comment:
Can we call this `_PWCFactor` ?

Reply:
Fixed in #374.

Reply:
I'd rather fix it here if possible to avoid mixing the PRs, is that ok for you? There are still a few things to fix in this PR to handle caching properly, and it will conflict with the renaming in your PR if we don't do it here.

Reply:
Of course !